### PR TITLE
feat: implemented power function

### PR DESCRIPTION
**Pull request by shripada**
This PR contains 

- Implementation of power function in the math module

### CODE REVIEW

File: math.js [hunk 1]
```diff
@@ -28,3 +28,11 @@ function testAdd() {
 function subtract(a, b) {
   return a - b;
 }
+
+function power(a, b) {
```
Comment (himanshukesarwani-dev) [inline math.js:32]: Please add some js documentation for this function.

File: math.js [hunk 2]
```diff
@@ -28,3 +28,11 @@ function testAdd() {
...
+  let product = a;
```
Comment (himanshukesarwani-dev) [inline math.js:33]: - what will happen if a and b are zero? 
- please add some pre condition checks here.

**Review by himanshukesarwani-dev — CHANGES_REQUESTED** [review]

File: math.js (hunk 1 above)
Comment (shripada) [reply math.js:38]: resolved

File: math.js (hunk 2 above)
Comment (shripada) [reply math.js:46]: resolved

File: math.js [hunk 3]
```diff
@@ -28,3 +28,24 @@ function testAdd() {
...
   return a - b;
 }
+
+/**
+ * This will compute the result by raising number a to the power of b in decimal system.
+ * @param {number} a the base number
+ * @param {number} b the power value to raise.
+ * @returns The result.
+ */
+function power(a, b) {
```
Comment (nikhilsnayak) [inline math.js:38]: please use a descriptive variable names

File: math.js [hunk 4]
```diff
@@ -28,3 +28,24 @@ function testAdd() {
...
+  // pre conditions
+  assert(typeof a === 'number');
+  assert(typeof b === 'number');
+  if (a === 0 || b === 0) {
+    return 0;
+  }
+
+  let product = a;
```
Comment (nikhilsnayak) [inline math.js:46]: - Initializing product with a results in bug.
- Instead initalizing product to 1 might resolve this bug

**Review by nikhilsnayak — CHANGES_REQUESTED** [review]

File: math.js (hunk 1 above)
Comment (himanshukesarwani-dev) [reply math.js:38]: Can you please improve the names of the arguments. a and b are too generic.

File: math.js [hunk 5]
```diff
@@ -35,16 +35,16 @@ function subtract(a, b) {
...
  * @returns The result.
  */
-function power(a, b) {
+function power(base, exponent) {
   // pre conditions
-  assert(typeof a === 'number');
-  assert(typeof b === 'number');
-  if (a === 0 || b === 0) {
+  assert(typeof base === 'number');
+  assert(typeof exonent === 'number');
```
Comment (nikhilsnayak) [inline math.js:41]: There is typo in exponent

**Review by nikhilsnayak — CHANGES_REQUESTED** [review]

File: math.js (hunk 1 above)
Comment (shripada) [reply math.js]: resolved

File: math.js (hunk 3 above)
Comment (shripada) [reply math.js]: resolved

File: math.js (hunk 4 above)
Comment (shripada) [reply math.js]: resolved

File: math.js (hunk 5 above)
Comment (shripada) [reply math.js]: resolved

**Review by nikhilsnayak — APPROVED** [review]
**LGTM**

**Review by himanshukesarwani-dev — APPROVED** [review]
LGTM

**Pull request merged**